src/filter/mod.rs

```diff
@@ -274,6 +276,24 @@ fn apply_to_commit2(
         Op::Squash => {
             return Some(history::rewrite_commit(repo, commit, &[], &commit.tree()?)).transpose()
         }
+        Op::Linear => {
+            let p: Vec<_> = commit.parents().collect();
+            if p.len() == 0 {
+                return Ok(Some(commit.id()));
+            }
+            let parent = some_or!(apply_to_commit2(op, &p[0], transaction)?, {
+                return Ok(None);
+            });
```

Comment:
I dont understand this case

Reply:
Returning `Ok(None)` from this function means that we cannot produce filtered commit right now because one of the parents was not filtered yet. In order to not recuruse here to filter the parent we return and will get called again at a later point when the parent has been filtered.